File: frontend/src/types/files.ts
export interface FileInfo {
  name: string;
  size: number;
  content_type: string;
  relative_path: string;
  modified_at: string;
  is_artifact: boolean;
}

export interface FileListResponse {
  files: FileInfo[];
  total_size: number;
}

export interface FileUploadResponse {
  uploaded: FileInfo[];
}

File: backend/app/schemas/files.py
"""Schemas for file management endpoints."""

from datetime import datetime

from pydantic import BaseModel


class FileInfo(BaseModel):
    """Metadata about a single file in a session workspace."""

    name: str
    size: int
    content_type: str
    relative_path: str
    modified_at: datetime
    is_artifact: bool

    model_config = {"from_attributes": True}


class FileListResponse(BaseModel):
    """Response for listing files in a session workspace."""

    files: list[FileInfo]
    total_size: int


class FileUploadResponse(BaseModel):
    """Response after uploading files."""

    uploaded: list[FileInfo]
